lib/rubocop/cop/performance/parallel_assignment.rb

```diff
@@ -0,0 +1,80 @@
+# encoding: utf-8
+
+module RuboCop
+  module Cop
+    module Performance
+      # Checks for simple usages of parallel assignment.
+      # This will only complain when the number of variables
+      # being assigned matched the number of assigning variables.
+      #
+      # @example
+      #   # bad
+      #   a, b, c = 1, 2, 3
+      #   a, b, c = [1, 2, 3]
+      #
+      #   # good
+      #   one, two = *foo
+      #   a, b = foo()
+      #
+      #   a = 1
+      #   b = 2
+      #   c = 3
+      class ParallelAssignment < Cop
+        MSG = 'Do not use parallel assignment.'
+
+        def on_masgn(node)
+          left, right = *node
+          left_elements = *left
+          right_elements = *right
+
+          # only complain when the number of variables matches
+          return if left_elements.compact.size != right_elements.compact.size
+
+          # allow mass assignment as the return of a method call
+          return if right.block_type? || right.send_type?
+
+          # allow mass assignment when using splat
+          return if (left_elements + right_elements).compact.any?(&:splat_type?)
+
+          return if left_elements.any? do |le|
+            right_elements.any? do |re|
+              re.respond_to?('loc') &&
+              re.loc.expression.is?(le.loc.expression.source)
+            end
+          end
+
+          add_offense(node, :expression)
+        end
+
+        def autocorrect(node)
+          left, right = *node
+
+          lambda do |corrector|
+            l_vars = extract_sources(left)
+            r_vars = extract_sources(right)
+            groups = l_vars.zip(r_vars)
+
+            assignment = groups.map do |pair|
+              pair[1] = 'nil' if pair[1].nil?
```

Comment:
This line is not covered by specs, i.e., it can be removed and everything still passes.